### PR TITLE
Add support for page scrolling in Playlist

### LINKED ISSUE #11: Unable to scroll past first page of songs in playlist
After selecting a playlist, I can only view the first 30 or so songs and can't scroll beyond that. Scrolling using vim keys or arrow keys brings me back to the top of the list, and ctl+d/u doesn't do anything (although I'm not too sure ctl+d/u should do). Hopefully I'm not missing something easy in the help section.

### PR DESCRIPTION
**Pull request by ghost**
Fix #11 

I tried to implement page scrolling in the Playlist and fixed `get_current_user_saved_tracks` that I pointed out in #128 .

I added and changed filed and methods of `App` but I'm not sure whether these changes are matched to what you expected. I would like to hear your opinion.

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -219,7 +219,8 @@ pub struct App {
     pub large_search_limit: u32,
     pub library: Library,
     pub playback_params: PlaybackParams,
-    pub playlist_tracks: Vec<PlaylistTrack>,
+    pub playlist_tracks: ScrollableResultPages<Page<PlaylistTrack>>,
```
Comment (Rigellute) [inline src/app.rs]: This is what I was thinking 👍

**Review by Rigellute — APPROVED** [review]
Great job again @echoSayonara.

This looks good to me 👍 

Perhaps once pagination has been implemented for all other views, a cleaner abstraction might be revealed.

**Comment by Rigellute** [comment]
@echoSayonara I've reverted the merge as playback doesn't seem to quite work when trying to play tracks from other pages.

Do you want to have a look?

**Comment by Rigellute** [comment]
I have a quick look and came up with this https://github.com/Rigellute/spotify-tui/compare/revert-133-page-scroll-rigel?expand=1

One issue here is that after the selection index gets above ~23, there is an "off-by-one" problem i.e. you press enter on the track, but the next one starts playing.

Could this be a problem with `usize` arithmetic?

**Comment by echoSayonara** [comment]
Looks good.
I checked "off-by-one" error and it did not occur.

However, I found another problem.
After open one playlist, I open another playlist and press enter on the track, but a track in other page was played. It seems page index is shared with all playlists.

**Comment by bdavs3** [comment]
**System**: MacOS | **Terminal**: KittyTerm + Zsh

@Rigellute I just installed `spotify-tui` for the first time. What I'm seeing with page scroll behavior in playlists is that pressing ctrl-d skips a page when it shouldn't. For instance:

* If I press `ctrl-d` on a playlist with 15 tracks, I can never access tracks 11-15
* If I press `ctrl-d` on a playlist with 45 tracks, I will skip from tracks 1-10 to tracks 21-30

Do you see this behavior as well? Want me to try to fix it?

**Comment by bdavs3** [comment]
Looks like this is being talked about in #362 actually

**Pull request merged**